plugins/keycloak_oidc_login/OidcLogin.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace app\plugins\keycloak_oidc_login;
+
+use Jumbojett\OpenIDConnectClient;
+use app\components\{LoginProviderInterface, RequestContext};
+use app\models\db\User;
+use app\models\settings\AntragsgruenApp;
+
+class OidcLogin implements LoginProviderInterface
+{
+
+    public function __construct(private string $issuerUrl, private string $clientId, private string $clientSecret){}
+
+    // map KC Claims to User attributes
+    private const PARAM_EMAIL = 'email';
+    private const PARAM_USERNAME = 'preferred_username';
+    private const PARAM_GIVEN_NAME = 'given_name';
+    private const PARAM_FAMILY_NAME = 'family_name';
```

Comment:
This mapping might be useful  in config as well - but it should be constant for keycloak, not so for other oidc providers. This and the 3 credentials should be the only things to change for a general OICD Plugin.

Reply:
Up to you. But I personally would probably not make things over-generic until the need arises :-)